rllib/policy/eager_tf_policy.py

```diff
@@ -669,14 +534,68 @@ def compute_log_likelihoods(
             return log_likelihoods
 
         @override(Policy)
-        def apply_gradients(self, gradients):
-            self._apply_gradients(
-                zip([(tf.convert_to_tensor(g) if g is not None else None)
-                     for g in gradients], self.model.trainable_variables()))
+        def postprocess_trajectory(self,
+                                   sample_batch,
+                                   other_agent_batches=None,
+                                   episode=None):
+            assert tf.executing_eagerly()
+            # Call super's postprocess_trajectory first.
+            sample_batch = Policy.postprocess_trajectory(self, sample_batch)
+            if postprocess_fn:
+                return postprocess_fn(self, sample_batch, other_agent_batches,
+                                      episode)
+            return sample_batch
 
+        @with_lock
         @override(Policy)
-        def get_exploration_state(self):
-            return convert_to_numpy(self.exploration.get_state())
+        def learn_on_batch(self, postprocessed_batch):
+            # Callback handling.
+            learn_stats = {}
+            self.callbacks.on_learn_on_batch(
+                policy=self,
+                train_batch=postprocessed_batch,
+                result=learn_stats)
+
+            if not isinstance(postprocessed_batch, SampleBatch) or \
+                    not postprocessed_batch.zero_padded:
+                pad_batch_to_sequences_of_same_size(
+                    postprocessed_batch,
+                    max_seq_len=self._max_seq_len,
+                    shuffle=False,
+                    batch_divisibility_req=self.batch_divisibility_req,
+                    view_requirements=self.view_requirements,
+                )
+
+            self._is_training = True
+            postprocessed_batch = self._lazy_tensor_dict(postprocessed_batch)
+            postprocessed_batch.is_training = True
```

Comment:
Absolutely, I was actually worried about this as well, since it's not prone to the `get_interceptor` of the `SampleBatch` (not a real key). Will make sure this is solid. Probably a good example to run here with eager-tracing=True is the BatchNormModel stuff.

Reply:
done